examples/Automatic Machine Learning Checkpoint.ipynb

```diff
@@ -0,0 +1,182 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import shutil\n",
+    "import time\n",
+    "\n",
+    "import numpy as np\n",
+    "\n",
+    "from neuraxle.base import BaseStep, NonFittableMixin\n",
+    "from neuraxle.checkpoints import DefaultCheckpoint\n",
+    "from neuraxle.hyperparams.distributions import RandInt\n",
+    "from neuraxle.hyperparams.space import HyperparameterSpace\n",
+    "from neuraxle.metaopt.random import RandomSearch\n",
+    "from neuraxle.pipeline import ResumablePipeline, DEFAULT_CACHE_FOLDER, Pipeline\n",
+    "\n",
+    "shutil.rmtree(DEFAULT_CACHE_FOLDER)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Multiplication(NonFittableMixin, BaseStep):\n",
+    "    def __init__(self, sleep_time=0.010, hyperparams=None, hyperparams_space=None):\n",
+    "        BaseStep.__init__(self, hyperparams=hyperparams, hyperparams_space=hyperparams_space)\n",
+    "        self.sleep_time = sleep_time\n",
+    "\n",
+    "    def transform(self, data_inputs):\n",
+    "        time.sleep(self.sleep_time)\n",
+    "        if not isinstance(data_inputs, np.ndarray):\n",
+    "            data_inputs = np.array(data_inputs)\n",
+    "\n",
+    "        return data_inputs * self.hyperparams['hp_mul']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def mean_squared_error(actual_outputs, expected_outputs):\n",
+    "    mses = [(a - b) ** 2 for a, b in zip(actual_outputs, expected_outputs)]\n",
+    "    return sum(mses) / float(len(mses))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def run_random_search(p, hyperparams_space, data_inputs, expected_outputs):\n",
```

Comment:
Useless argument `hyperparams_space`. It could be pre-setted to the pipeline before at its definition. E.g.: 

```
pipeline = Pipeline([
    ('multiplication_2', Multiplication(sleep_time=0.01)),
    ...
    ('multiplication_11', Multiplication(sleep_time=0.01)),
]).set_hyperparams_space(hyperparams_space)
```

Reply:
Because the setter method returns self like a builder pattern.